### PR TITLE
Create rules_overview.md

### PR DESCRIPTION
**Pull request by spacemanspiff2007**
Based on the discussion in the forums this should serve as a starting point and show what technologies are available.
I did not add advantages/disadvantages because they are often very subjective and vary from person to person.
It's more meant as a starting point for further research.

Should I also include links to the respective docs?
What do you think?

### CODE REVIEW

**Comment by rkoshak** [comment]
I'm not yet convinced the Getting Started Tutorial is the best place to put this. As part of #1855 I'm planning on adding a top level Rules section to the right hand docs menu (similar to how UIs has its own section). My thought is that we would have an intro page there and the table would go there and the getting started tutorial would link to that. 

But maybe it makes sense to put it in Getting Started for now, and then move it later as #1855 matures. 

Or, maybe it makes sense to add it to the new [Rules Concepts Page](https://next.openhab.org/docs/concepts/rules.html), just before the comprehensive examples.

I also envision a more comprehensive table. Something like:

Name | Language | External | UI Support | Library Support | Helper Library | Notes
-|-|-|-|-|-|-
add-on name as a link to it's doc | Language: programming language and version supports | HABApp and Node Red would have a check, the rest blank | Instead of talking about whether or not the add-on/engine is UI or not, my concern is wether the add-on works and supports well Script Actions and Script Conditions in the UI or not. For HABApp and Node Red this would be NA. I largely want to capture if there are any add-ons that explicitly will not work in that context (e.g. the Java add-on on the marketplace does not support the UI at all).  | Whether or not the language easily supports third part libraries. I'll need help with that but I think Nashorn and Jython would be "limited", I'm not sure about Groovy, JS Scripting would say "most npm libraries", and jRuby "most Ruby gems". | Whether or not a helper library exists, included with the add-on or third party installable, link to docs, etc. NodeRed and HABApp would have "NA". | Here is where we'd capture stuff like "Rules DSL is not a great choice if just getting started", "Jython support is limited", etc.

If we go forward with this PR as is for now, I'll open a review because there are a couple of inaccuracies.

File: tutorials/getting_started/rules_overview.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+|---------------------|--------------------|---------------------|-------------------------------|
+| Blockly             | :heavy_check_mark: | Graphical           | Only available through the UI |
+| GraalVM JavaScript  | :heavy_check_mark: | JavaScript latest   | breaks Blockly/Nashorn rules  |
+| Groovy              | :heavy_check_mark: | Groovy              |                               |
+| HABApp              | :x:                | >= Python 3.8       |                               |
+| Java                | :heavy_check_mark: | Java                |                               |
+| jRuby               | :heavy_check_mark: | Ruby                |                               |
+| Jython              | :heavy_check_mark: | Python 2.7          |                               |
+| NodeRed             | :x:                | Graphical           | Own UI                        |
+| Nashorn JavaScript  | :heavy_check_mark: | JavaScript 5.1      |                               |
```
Comment (rkoshak) [inline tutorials/getting_started/rules_overview.md:26]: Let's use the official ECMAScript 5.1. Our friends over at Oracle have JavaScript trademarked.

**Review by rkoshak — COMMENTED** [review]
One inaccuracy and some suggestions.

**Comment by Confectrician** [comment]
> I'm not yet convinced the Getting Started Tutorial is the best place to put this.

I would agree for this point.
Getting started may be a place to link to the rules overview, but not for a complete explanation, which probably will cause confusion.

File: tutorials/getting_started/rules_overview.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+The list is sorted alphabetically by name and it is meant as a starting point for further research.
+
+| Name                | Native             | Language/Graphical  | Notes                         |
+|---------------------|--------------------|---------------------|-------------------------------|
+| Blockly             | :heavy_check_mark: | Graphical           | Only available through the UI |
+| GraalVM JavaScript  | :heavy_check_mark: | JavaScript latest   | breaks Blockly/Nashorn rules  |
+| Groovy              | :heavy_check_mark: | Groovy              |                               |
+| HABApp              | :x:                | >= Python 3.8       |                               |
+| Java                | :heavy_check_mark: | Java                |                               |
```
[the remark this answers is not in the document]
Comment (spacemanspiff2007) [reply tutorials/getting_started/rules_overview.md]: I have no strong feeling about it. Should I remove it?

**Comment by spacemanspiff2007** [comment]
My initial thought was too keep the list simple so it's just an overview but show it somewhere in "getting started".
E.g. if I already know groovy I can go to the docs there instead of doing all of the blockly tutorial.
I envision a link to a more complete overview like Rich suggested where there is an in-depth comparison (on another site).

**Comment by Confectrician** [comment]
I would like to get this into the docs for the next stable version website.
So i would suggest to remove the java rule table row and merge it in the current location.

We just should keep an eye on it and go on with further improving after the openHAB release.

**Review by Confectrician — APPROVED** [review]
Thanks. 👍🏼

**Pull request merged**